scripts/stablex/utilities.js

```diff
@@ -143,6 +143,36 @@ const sendLiquidityOrders = async function(
   )
 }
 
+// Given an orderbook A:B & B:C compute orderbook B:C
+const transitiveOrderbook = function(lhs, rhs) {
+  const result = { bids: [], asks: [] }
+  const lhs_asks = lhs.asks.values()
+  const rhs_asks = rhs.asks.values()
+
+  let rhs_next = rhs_asks.next()
+  let lhs_next = lhs_asks.next()
+  while (!(lhs_next.done || rhs_next.done)) {
+    const price = lhs_next.value[0] * rhs_next.value[0]
```

Comment:
nit: I personally find it easier to read if you create a constant with a semantic name for `lhs_next.value[0]` or `lhs_next.value[1]` and include some extra comments in plain english for example for the if predicate or their branches.